diff --git a/CMakeLists.txt b/CMakeLists.txt
index a3eb0e6..dc4ed45 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,13 @@
 cmake_minimum_required(VERSION 3.9)
 project(cmdparser)

+# Create a header only library and explicitly specify the include directory
+# for consumers of the library
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(${PROJECT_NAME} INTERFACE
+  "${CMAKE_CURRENT_SOURCE_DIR}"
+)
+
 enable_testing()

 set(CMAKE_CXX_STANDARD 11)
